tests/base.go

```diff
@@ -114,15 +110,24 @@ func RandomName() string {
 
 	return name
 }
+func IPFromCidr(cidr string) (string, error) {
+	ip, _, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return "", err
+	}
+	return ip.String(), nil
+}
 
-func Wait(addr string, port string) bool {
+func Wait(addr string, port string) error {
+	var err error
 	for t := time.Now(); time.Since(t) < 3*time.Minute; {
-		_, err := net.DialTimeout("tcp", net.JoinHostPort(addr, "22"), time.Second*12)
+		con, err := net.DialTimeout("tcp", net.JoinHostPort(addr, port), time.Second*12)
+		con.Close()
```

Comment:
i think `con` should be closed only if `err` is `nil` because it panics if there is an error and `con` is `nil`